src/fhirtypes/ElementDefinition.ts

```diff
@@ -425,6 +428,39 @@ export class ElementDefinition {
       }
     }
 
+    const connectedElements = this.findConnectedElements();
+    if (connectedElements.length > 0) {
+      // check to see if the card constraint would actually be a problem for the connected element
+      // that is to say, if the new card is narrower than the connected card
+      connectedElements.forEach(ce => {
+        if ((ce.min == null && min > 0) || (ce.min != null && ce.min < min)) {
```

Comment:
So, working through this scenario: if this were allowed, then with one-sided cardinality rules, we could end up with something like:
```
* component[Lab].interpretation ..9
* component.interpretation 2..9
```
But, looking more closely at this case: since `component[Lab].interpretation.min` is `null`, then it should end up inheriting the minimum of 2 from `component.interpretation`, which is fine. So, I think you're right: if the min (or max, as you point out in your next comment) is null, then that's saying "we'll inherit the value," which is always fine.

It does mean it may be necessary to check for a scenario like this:
```
* component[Lab].interpretation ..9 // or other rule that creates this element
* component.interpretation 2..9
* component[Lab].interpretation 1..9
```
The third rule should not be applied in this case, since it would mean that going inside the `Lab` slice widens the cardinality of `interpretation`. If the first rule is skipped, then existing code should handle the error, since then the `component[Lab].interpretation` element will get created with the copied cardinality, I think.

Reply:
I don't think this necessarily makes it so that your second example wouldn't be an issue. But one thing to consider is that one sided cardinalities don't set the cardinality to null, they just get it from the element's current cardinality. So in the second example I think `component[Lab].interpretation` would maintain whatever min cardinality was originally on `component.interpretation`.